mailbox/api/src/main/java/org/apache/james/mailbox/model/MailboxQuery.java

```diff
@@ -136,20 +128,43 @@ private MailboxPath buildBase() {
      * @param pathDelimiter
      *            path delimiter to use
      */
-    @VisibleForTesting MailboxQuery(MailboxPath base, String expression, char pathDelimiter) {
-        super();
-        this.base = base;
-        if (base.getName() == null)
-            this.base.setName("");
+    @VisibleForTesting MailboxQuery(Optional<String> namespace, Optional<String> user, Optional<String> name,
+                                    String expression, MailboxSession session) {
+        this.namespace = namespace;
+        this.user = user;
+        this.name = name;
+        this.base = new MailboxPath(
+            namespace.orElse(MailboxConstants.USER_NAMESPACE),
+            user.orElse(session.getUser().getUserName()),
+            name.orElse(EMPTY_PATH_NAME));
         if (expression == null) {
             this.expression = "";
         } else {
             this.expression = expression;
         }
-        this.pathDelimiter = pathDelimiter;
+        this.pathDelimiter = session.getPathDelimiter();
         pattern = constructEscapedRegex();
     }
 
+    public MailboxPath getPathLike() {
+        String combinedName = getCombinedName()
+            .replace(getFreeWildcard(), SQL_WILDCARD_CHAR)
+            .replace(getLocalWildcard(), SQL_WILDCARD_CHAR)
```

Comment:
With the current value of the constant this line does strictly nothing

Reply:
idem